docs/content/using-npm/scripts.md

```diff
@@ -10,90 +10,64 @@ description: How npm handles the "scripts" field
-* **publish**, **postpublish**:
-  Run AFTER the package is published.
-* **preinstall**:
-  Run BEFORE the package is installed
-* **install**, **postinstall**:
-  Run AFTER the package is installed.
-* **preuninstall**, **uninstall**:
-  Run BEFORE the package is uninstalled.
-* **postuninstall**:
-  Run AFTER the package is uninstalled.
-* **preversion**:
-  Run BEFORE bumping the package version.
-* **version**:
-  Run AFTER bumping the package version, but BEFORE commit.
-* **postversion**:
-  Run AFTER bumping the package version, and AFTER commit.
-* **pretest**, **test**, **posttest**:
-  Run by the `npm test` command.
-* **prestop**, **stop**, **poststop**:
-  Run by the `npm stop` command.
-* **prestart**, **start**, **poststart**:
-  Run by the `npm start` command.
-* **prerestart**, **restart**, **postrestart**:
-  Run by the `npm restart` command. Note: `npm restart` will run the
-  stop and start scripts if no `restart` script is provided.
-* **preshrinkwrap**, **shrinkwrap**, **postshrinkwrap**:
-  Run by the `npm shrinkwrap` command.
-
-Additionally, arbitrary scripts can be executed by running `npm
-run-script <stage>`. *Pre* and *post* commands with matching
-names will be run for those as well (e.g. `premyscript`, `myscript`,
-`postmyscript`). Scripts from dependencies can be run with 
-`npm explore <pkg> -- npm run <stage>`.
-
-#### Prepublish and Prepare
-
-#### Deprecation Note
-
-Since `npm@1.1.71`, the npm CLI has run the `prepublish` script for both `npm
-publish` and `npm install`, because it's a convenient way to prepare a package
-for use (some common use cases are described in the section below).  It has
-also turned out to be, in practice, [very
-confusing](https://github.com/npm/npm/issues/10074).  As of `npm@4.0.0`, a new
-event has been introduced, `prepare`, that preserves this existing behavior. A
-_new_ event, `prepublishOnly` has been added as a transitional strategy to
-allow users to avoid the confusing behavior of existing npm versions and only
-run on `npm publish` (for instance, running the tests one last time to ensure
-they're in good shape).
-
-See <https://github.com/npm/npm/issues/10074> for a much lengthier
-justification, with further reading, for this change.
-
-#### Use Cases
-
-If you need to perform operations on your package before it is used, in a way
-that is not dependent on the operating system or architecture of the
-target system, use a `prepublish` script.  This includes
-tasks such as:
+The `"scripts"` property of of your `package.json` file supports a number of built-in scripts and their preset life cycle events as well as arbitrary scripts. These all can be executed by running `npm run-script <stage>` or `npm run <stage>` for short. *Pre* and *post* commands with matching names will be run for those as well (e.g. `premyscript`, `myscript`, `postmyscript`). Scripts from dependencies can be run with `npm explore <pkg> -- npm run <stage>`.
```

Comment:
it might be worth noting that although "test" can be user-defined/overridden, "install" can't be (and probably others, like "publish" etc)?

Reply:
That's a great point, same with `pack`. I'll add a "caveats" section. Does that sound right?

Reply:
Rather than a caveats section I think what I mean is that we have some shorthands for scripts.

`npm run test` -> `npm test`
`npm run start` -> `npm start`
`npm run build` -> `npm build`
`npm run restart` -> `npm restart`

I can't think of any others, can you @ljharb ?

Reply:
Shorthands that *can* be overridden for user scripts would also include `version`, but I can't think of any others.

It'd be great to exhaustively list BOTH sets - ie, one list of "shorthand commands that run user scripts when present" and "shorthand commands that ignore user scripts even if present" (noting that they all invoke pre/post scripts)